### PR TITLE
feat: Add more thread functions

### PR DESCRIPTION
**Pull request by mAxYoLo01**
## About

Added:
- `Guild.get_all_active_threads(self) -> List[Thread]`
- `Channel.remove_member(self, member_id: Union[int, Snowflake, "Member"]) -> None`
- `Channel.get_members(self) -> List[ThreadMember]`
- `Channel.leave_thread(self) -> None`
- `Channel.join_thread(self) -> None`

## Checklist

- [x] I've ran `pre-commit` to format and lint the change(s) made.
- [x] I've checked to make sure the change(s) work on `3.8.6` and higher.
- [ ] This fixes/solves an [Issue](https://github.com/goverfl0w/discord-interactions/issues).
  - (If existent):
- [ ] I've made this pull request for/as: (check all that apply)
  - [ ] Documentation
  - [ ] Breaking change
  - [x] New feature/enhancement
  - [ ] Bugfix

### CODE REVIEW

File: interactions/api/models/guild.py [hunk 1]
```diff
@@ -1677,6 +1677,26 @@ async def get_all_channels(self) -> List[Channel]:
...
+        :rtype: List[Thread]
+        """
+        if not self._client:
+            raise LibraryException(code=13)
+        res = await self._client.list_active_threads(int(self.id))
+        threads = [Thread(**thread, _client=self._client) for thread in res["threads"]]
+        members = [ThreadMember(**member, _client=self._client) for member in res["members"]]
+        for member in members:
+            for thread in threads:
+                if int(thread.id) == int(member.id):
```
Comment (sillyElvy) [inline interactions/api/models/guild.py:1694]: wait thread IDs are member IDs?

Comment (mAxYoLo01) [reply interactions/api/models/guild.py:1694]: > wait thread IDs are member IDs?

no, it's just that a `ThreadMember` has two IDs
- id: ID of the thread
- user_id: ID of the user

Comment (sillyElvy) [reply interactions/api/models/guild.py:1694]: Ah alr

File: interactions/api/models/channel.py [hunk 2]
```diff
@@ -1126,6 +1147,43 @@ async def get_webhooks(self) -> List[Webhook]:
...
+        """
+        if not self._client:
+            raise LibraryException(code=13)
+        if not self.thread_metadata:
+            raise LibraryException(message="The Channel you specified is not a thread!", code=12)
+
+        res = await self._client.list_thread_members(int(self.id))
+        return [ThreadMember(**member, _client=self._client) for member in res]
+
+    async def leave_thread(self) -> None:
```
Comment (sillyElvy) [inline interactions/api/models/channel.py]:
```suggestion
    async def leave(self) -> None:
```
Remove redundant naming

File: interactions/api/models/channel.py [hunk 3]
```diff
@@ -1126,6 +1147,43 @@ async def get_webhooks(self) -> List[Webhook]:
...
+        Removes the bot from the thread
+        """
+        if not self._client:
+            raise LibraryException(code=13)
+        if not self.thread_metadata:
+            raise LibraryException(message="The Channel you specified is not a thread!", code=12)
+
+        await self._client.leave_thread(int(self.id))
+
+    async def join_thread(self) -> None:
```
Comment (sillyElvy) [inline interactions/api/models/channel.py]:
```suggestion
    async def join(self) -> None:
```
Remove redudant naming

File: interactions/api/models/channel.pyi [hunk 4]
```diff
@@ -161,6 +162,9 @@ class Channel(ClientSerializerMixin):
     ) -> Invite: ...
     async def get_history(self, limit: int = ...) -> List[Message]: ...
     async def get_webhooks(self) -> List[Webhook]: ...
+    async def get_members(self) -> List[ThreadMember]: ...
+    async def leave_thread(self) -> None: ...
```
Comment (sillyElvy) [inline interactions/api/models/channel.pyi]:
```suggestion
    async def thread(self) -> None: ...
```

File: interactions/api/models/channel.pyi [hunk 5]
```diff
@@ -161,6 +162,9 @@ class Channel(ClientSerializerMixin):
...
+    async def join_thread(self) -> None: ...
```
Comment (sillyElvy) [inline interactions/api/models/channel.pyi]:
```suggestion
    async def join(self) -> None: ...
```

File: interactions/api/models/guild.py [hunk 6]
```diff
@@ -1677,6 +1677,26 @@ async def get_all_channels(self) -> List[Channel]:
...
+        """
+        Gets all active threads of the guild.
+
+        :return: The threads of the guild.
+        :rtype: List[Thread]
+        """
+        if not self._client:
+            raise LibraryException(code=13)
+        res = await self._client.list_active_threads(int(self.id))
+        threads = [Thread(**thread, _client=self._client) for thread in res["threads"]]
```
Comment (sillyElvy) [inline interactions/api/models/guild.py]: why not the full response?

**Review by sillyElvy — CHANGES_REQUESTED** [review]

File: interactions/api/models/guild.py (hunk 6 above)
Comment (mAxYoLo01) [reply interactions/api/models/guild.py]: What?

Comment (sillyElvy) [reply interactions/api/models/guild.py]: nvm I just looked at the returns of the endpoint, forget it

File: interactions/api/models/guild.py [hunk 7]
```diff
@@ -1677,6 +1677,26 @@ async def get_all_channels(self) -> List[Channel]:
         self.channels = [Channel(**channel, _client=self._client) for channel in res]
         return self.channels
 
+    async def get_all_active_threads(self) -> List[Thread]:
```
Comment (sillyElvy) [inline interactions/api/models/guild.py]: Can we make this return channel? according to https://discord.com/developers/docs/resources/guild#list-active-guild-threads, it is also a list of channels

Comment (mAxYoLo01) [reply interactions/api/models/guild.py]: Sure, so we really never use `Thread` object xD

**Review by sillyElvy — APPROVED** [review]
LGTM

**Pull request merged**